pandas/core/indexes/datetimelike.py

```diff
@@ -520,7 +520,8 @@ def where(self, cond, other=None):
             other = other.view("i8")
 
         result = np.where(cond, values, other).astype("i8")
-        return self._shallow_copy(result)
+        arr = type(self._data)._simple_new(result, dtype=self.dtype)
```

Comment:
why is _shallow_copy just not this exactly? it seems we have been expanding to these 2 lines rather than just using _shallow_copy.

Reply:
because we're being more careful/explicit about when/whether to retain `self.freq`.  e.g. until recently this method incorrectly retained self.freq for DTI/TDI.

Reply:
because we're being more careful/explicit about when/whether to retain `self.freq`.  e.g. until recently this method incorrectly retained self.freq for DTI/TDI.